js/common/util/compute_state_display.js

```diff
@@ -16,7 +16,7 @@ export default function computeStateDisplay(localize, stateObj, language) {
         stateObj._stateDisplay = localize(`state.${domain}.default.${stateObj.state}`);
       }
     } else if (stateObj.attributes.unit_of_measurement && !['unknown', 'unavailable'].includes(stateObj.state)) {
-      stateObj._stateDisplay = stateObj.state + ' ' + stateObj.attributes.unit_of_measurement;
+      stateObj._stateDisplay = `${stateObj.state}&nbsp${stateObj.attributes.unit_of_measurement}`;
```

Comment:
Does this work? I was under the impression that Polymer does not allow setting HTML.

I also think that this fix is too much upstream. We don't know where state display is going to be used. I think that it is better to fix it inside the component that will display the state.